packages/core/core/src/Validation.js

```diff
@@ -91,11 +97,30 @@ export default class Validation {
         });
       }
 
-      await validator.validate({
+      let validatorResult = await validator.validate({
         asset: new Asset(asset),
         options: pluginOptions,
         config
       });
+
+      if (validatorResult) {
+        let {warnings, errors} = validatorResult;
+
+        for (let error of errors) {
+          logger.error(error);
+        }
+
+        for (let warning of warnings) {
+          logger.warn(warning);
+        }
+
+        // Throw to fail build...
+        if (errors.length > 0) {
+          throw new ThrowableDiagnostic({
```

Comment:
Yeah... it would print twice was gonna refactor buildFailure and error handling a bit more in a follow-up, not entirely sure yet how to do it.

Arrays seem like a good idea